Sync changes for 2.25 release

VaishnaviHire opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Merge criteria:
<!--- This PR will be merged by any repository approver when it meets all the points in the checklist -->
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->

- [ ] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- New Features
  - LlamaStackDistribution now exposes an internal Service URL in status for easier discovery.

- Improvements
  - Container startup now uses a version-adaptive launcher so the server starts correctly across releases.
  - Sample images and provider IDs switched to the "starter" distribution; sample includes a 20Gi persistent storage example.

- Documentation
  - API docs and README updated to include serviceURL and new Go prerequisite.

- Tests
  - E2E tests parameterized by distribution and include readiness checks.

- Chores
  - CI/tooling updated to derive Go version from go.mod and bumped Go runtime.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

rhuss reviewed (approved):
lgtm

rhdedgar committed:
standardize DfFIPS settings (#147)

Updating relevant settings to make the same DfFIPS configurations on some other RHOAI components, for the purpose of passing [check-payload](https://github.com/openshift/check-payload/) scans without errors or false-positives.

- CGO settings updated
- Golang version incremented
- Some context / test changes  made to accommodate the Go version update.

Fixes: RHAIENG-663

If merged before 139, then bumps the UBI of the builder image, and:
Closes: #139

Approved-by: VaishnaviHire
(cherry picked from commit 7751adc00792db8de5b9619254ed3024f454604e)

zdtsw committed:
uplift(Go): version from 1.23 to 1.24 (#148)

- Go 1.23 is EOL
- fix linter: usetesting context.Background()
  - context.Background() create independent context but this is causing lint error but this wont have rate limit
  - t.Context() is shared by all testscases in the same test, this could cause rate limit by the client esp when many run in parallel. to add a 30seconds as a workaround to mitigate rate limit issue.
- uplift setup-go to use v5.5.0 across all workflow and use go.mod for Go version

~~DO NOT MERGE TILL https://github.com/llamastack/llama-stack-k8s-operator/pull/139 IS IN FIRST~~

Approved-by: rhdedgar
(cherry picked from commit f284cc83029832b6ef4ed75ebe458c764c418e65)

leseb committed:
fix: adapt the container entrypoint based on the llama-stack version (#153)

In llama-stack 0.2.17, with the help of
https://github.com/llamastack/llama-stack/pull/2975 the module name containing the server changed. So we need to adapt the entrypoint in consequence.

When using a ConfigMap to store the server configuration, the operator overrides the container entrypoint. This code has been adapted to use the correct module based on the detected llama-stack version.

Also, we now force the entrypoint to use python3 instead of python.

Approved-by: VaishnaviHire
(cherry picked from commit 68d3c6a47efa5889b2e37fbff629d36843510ad8)

nathan-weinberg committed:
chore: add CODEOWNERS file (#155)

The following people need appropriate access granted
```
    Unknown owner on line 5: make sure @cdoern exists and has write access to the repository
    Unknown owner on line 5: make sure @derekhiggins exists and has write access to the repository
    Unknown owner on line 5: make sure @Elbehery exists and has write access to the repository
    Unknown owner on line 5: make sure @skamenan7 exists and has write access to the repository
```

Signed-off-by: Nathan Weinberg <nweinber@redhat.com>
(cherry picked from commit a8e0fbf4195b75efdcdb047df751c2bffea17b42)

VaishnaviHire committed:
(fix): Fix Build Image workflow (#158)

This commit fixes build-image workflow. Recent [changes](#148 ) were made to identify Go version using go.mod file. This requires go.mod file to be available before setting up Go.

Workflow failure: https://github.com/llamastack/llama-stack-k8s-operator/actions/runs/17467245024

Approved-by: nathan-weinberg

Approved-by: rhdedgar

Approved-by: leseb
(cherry picked from commit e75b7be51bc0d8762c24ce8fd4b8e3785a4ea8eb)

VaishnaviHire committed:
Update base image for initContainers (#159)

Remove dependency on additional images in operator. This commit replaces base image for initContainers with main container's base image. This removes dependency of any additional images in disconnected environments

Approved-by: derekhiggins

Approved-by: rhdedgar

Approved-by: leseb
(cherry picked from commit 5775e80abc5bbe2f656f37b36159100d137a8b90)

derekhiggins committed:
test: remove ollama distribution testing, replace with starter (#160)

test: remove ollama distribution testing, replace with starter

    This focuses testing on the actively maintained starter distribution
    while removing support for the older ollama image that's no longer updated.

    Also intrduces a list of distro images (currently only 1) that can be tested.

---
    test: combine creation and deletion into single lifecycle test

    Restructure the creation test to run creation followed by deletion in a single
    test suite. This enables testing multiple distribution images sequentially
    with proper cleanup between runs.

    Changes:
    - Add TestCreationDeletionSuite that runs creation then deletion
    - Replace TestDeletionSuite with runDeletionTests helper function
    - Simplify test structure while maintaining all existing functionality

Approved-by: rhdedgar

Approved-by: leseb
(cherry picked from commit f65cb85f6a9a0f9bd9a4c3ba6c4820b4096601cc)

ChristianZaccaria committed:
Add Service Endpoint to LlamaStackDistribution .status (#162)

## Description

The `.status.serviceURL` field in the `LlamaStackDistribution` resource would now have the value i.e.,:
```
http://<service-name>.<namespace>.svc.cluster.local:<port>
```

Note: the pre-commit linter seems to not like the use of `serviceUrl`. It said to use `serviceURL` instead.

## Verification Steps
I tested the service by:
1. Running my own version of llama-stack-k8s-operator on OpenShift with the changes in this PR: `make deploy IMG=quay.io/rh_ee_czaccari/llama-stack-k8s-operator:servicestatus4`
2. Created a sample LlamaStackDistribution.
3. Then I ran a test pod:

>kubectl run curl-test --image=quay.io/rh_ee_czaccari/curl:test5 --rm -it --restart=Never -n ollama-dist -- curl -k -v http://llamastackdistribution-sample-service.ollama-dist.svc.cluster.local:8321/v1/models

Approved-by: leseb

Approved-by: derekhiggins
(cherry picked from commit f2e783ffccdc4c0e056c10890daa6a19cb0b38a8)

leseb commented on .github/workflows/main-build-image.yml:
```diff
@@ -13,14 +13,14 @@ jobs:
+      - name: Set up Go 1.24
+        uses: actions/setup-go@d35c59abb061a4a6fb18e82ac0862c26744d6ab5 # v5.5.0
+        with:
+          go-version: '1.23'
```
`Set up Go 1.24`

just spotted that :(

VaishnaviHire replied:
Fixed this

VaishnaviHire committed:
Fix ODH workflows

Fix odh image build workflows to match with upstream. Set go-version
from go.mid file

Signed-off-by: Vaishnavi Hire <vhire@redhat.com>

ChristianZaccaria reviewed (commented):
/lgtm from the RAG team - the requested commit on **Add Service Endpoint to LlamaStackDistribution .status** is found in this PR. Thank you!

VaishnaviHire committed:
Fix Linter errors

Signed-off-by: Vaishnavi Hire <vhire@redhat.com>
